text/0000-sync-networked-vars.md

```diff
@@ -0,0 +1,114 @@
+- Feature Name: `synced_networked_var`
+- Start Date: 2021-02-12
+- RFC PR: [Unity-Technologies/com.unity.multiplayer.rfcs#0000](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/pull/0000)
+- Issue: [Unity-Technologies/com.unity.multiplayer#0000](https://github.com/Unity-Technologies/com.unity.multiplayer/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Make the NetworkedVariables synchronized with the tick system.
+
+We suggest two elements:
+- Changing `NetworkedBehaviour::NetworkedBehaviourUpdate()` so that the function runs at most one per Network Tick. When called more than once per Network Tick, it would simply return.
+- NetworkedVariables carrying two pieces of information with their delta updates:
+    - The local tick at which a variable was modified
+    - The remote tick at which the variable was modified, when the host relays the variable to clients, including the source client    
+
+# Motivation
+[motivation]: #motivation
+
+In a client-server topology, any client and the server can modify NetworkedVariables. The server receives the clients' changes and send them back to all of the clients. In order to retain generality, we'll assume that any variable can be modified by anyone and that the server can also modify the variable as they are received and sent. For example, we could have server-driven physics over client-driven character movement.
```

Comment:
Would it be too restrictive to make the rule like this:
1) variables belong to a behavior
2) behaviors belong to an object
3) objects are owned by at most 1 client or the server
4) hence, only the owner (either one and only one client or the server) can write changes?
5) and objects can change ownership

Reply:
I think this is a design question: how granular ownership should be controlled? object vs behavior vs variable. so far, it sounds like netvar could be owned by either server or client (and even maybe both?). it sounds weird to me that a client could own an object and behaviour but not all netvars because one of them is owned/controlled by the server — but I can see a few use-cases that's why I'm fine with it.

in this context, netvar should not care which level ownership/control is applied and ask/query for that information from outside (hence my comment above suggesting to find out ownership by networkobject's ownerid lookup).

Reply:
hrm, I'm probably over-simplifying but I was thinking it would be all-or-nothing - an object is owned by a client or the server and hence all the variables go along for the ride.  @TwoTenPvP you've got lots of experience w/ customers here - are there cases worth doing work for where we want some but not all variables owned by the client within an object?